snuba/datasets/profiles_processor.py

```diff
@@ -9,44 +9,80 @@
+                        "model",
+                    ]
+                }
+                additional_fields["architecture"] = message["device"].get(
+                    "architecture", "unknown"
+                )
+                additional_fields["version_name"] = message["release"]
+
+                for field in ["name", "version", "build_number"]:
+                    additional_fields[f"device_os_{field}"] = message["os"].get(field)
+
+                transaction = message["transactions"][0]
+                additional_fields.update(
+                    {
+                        "transaction_id": transaction["id"],
+                        "transaction_name": transaction["name"],
+                        "trace_id": transaction["trace_id"],
+                        "duration_ns": int(
+                            transaction["relative_end_ns"]
+                            - transaction["relative_start_ns"]
+                        ),
+                        "profile_id": str(UUID(message["event_id"])),
+                    }
+                )
+            else:
+                additional_fields = {
+                    f"device_{field}": message.get(f"device_{field}", "")
+                    for field in ["locale", "classification", "manufacturer", "model"]
+                }
+                additional_fields["architecture"] = message.get(
+                    "architecture", "unknown"
+                )
+                additional_fields["version_name"] = message["version_name"]
+                additional_fields["device_os_name"] = message["device_os_name"]
+                additional_fields["device_os_version"] = message["device_os_version"]
+                additional_fields["duration_ns"] = message["duration_ns"]
+                additional_fields["transaction_name"] = message["transaction_name"]
+                additional_fields["transaction_id"] = str(
+                    UUID(message["transaction_id"])
+                )
+                additional_fields["trace_id"] = str(UUID(message["trace_id"]))
+                additional_fields["profile_id"] = str(UUID(message["profile_id"]))
+
             processed = {
-                "organization_id": message["organization_id"],
-                "project_id": message["project_id"],
-                "transaction_id": str(UUID(message["transaction_id"])),
-                "profile_id": str(UUID(message["profile_id"])),
-                "received": datetime.utcfromtimestamp(message["received"]),
-                "profile": message["profile"],
                 "android_api_level": message.get("android_api_level"),
-                "device_classification": message["device_classification"],
-                "device_locale": message["device_locale"],
-                "device_manufacturer": message["device_manufacturer"],
-                "device_model": message["device_model"],
                 "device_os_build_number": message.get("device_os_build_number"),
```

Comment:
I think you'll want to move `device_os_build_number` into the else block since the sampled format overwrites this value.